assemblers/org.osbuild.qemu

```diff
@@ -332,33 +332,40 @@ def install_grub2(image: str, pt: PartitionTable):
     """Install grub2 to image"""
     grub2_core = "/var/tmp/grub2-core.img"
 
+    # Create the level-2 & 3 stages of the bootloader, aka the core
+    # it consists of the kernel plus the core modules required to
+    # to locate and load the rest of the grub modules, specifically
+    # the "normal.mod" (Stage 4) module.
+    # The exact list of modules required to be built into the core
+    # depends on the system: it is the minimal set needed to find
+    # read the partition and its filesystem containing said modules
+    # and the grub configuration [NB: efi systems work differently]
+
+    # modules: access the disk and read the partition table
+    modules = ["biosdisk", "part_msdos"]
```

Comment:
Indeed, the tests verify the checksum of grub's level 2 stage, and that changes based on the order the modules are passed.

We could fix this by updating the checksum (boot tests pass, so we should be good), or by rearranging the code to produce the same order as before.

Reply:
This is weird because I am quite certain that the `diskboot` image will be the first one to be written to the core image anyway and the order of the other two (`part_msdos`, `fs_module`) should not have changed. Let me have a look at the grub source.

Reply:
I seem to be right about [`diskboot`](https://github.com/rhboot/grub2/blob/4dd4ceec023111a4ccf69f8de6fa0885c6847a35/util/mkimage.c#L1217) but of course `biosdisk` is a different matter (I got confused sorry). Other modules, including said `biosdisk`, seem to be indeed [added](https://github.com/rhboot/grub2/blob/4dd4ceec023111a4ccf69f8de6fa0885c6847a35/util/mkimage.c#L956) as specified (after dependencies got [resolved](https://github.com/rhboot/grub2/blob/4dd4ceec023111a4ccf69f8de6fa0885c6847a35/util/mkimage.c#L845). I thought disk, partition, filesystem is quite the natural order, but I can also change it back to the original one. Opinions?

Reply:
I had forgotten exactly what the issue was, but as this seemingly was just about the hash in the tests, changing it around and updating the tests is perfectly fine with me.